src/noggit/texture_set.cpp

```diff
@@ -946,24 +979,35 @@ namespace
 bool TextureSet::apply_alpha_changes()
 {
   if (!tmp_edit_values || nTextures < 2)
-  {
-    tmp_edit_values = boost::none;
-    return false;
-  }
-
-  auto& new_amaps = tmp_edit_values.get();
+    {
+        tmp_edit_values = boost::none;
+        return false;
+    }
 
-  for (int alpha_layer = 0; alpha_layer < nTextures - 1; ++alpha_layer)
-  {
-    std::array<std::uint8_t, 64 * 64> values;
+    auto& new_amaps = tmp_edit_values.get();
 
+    std::array<std::uint8_t, 64 * 64> values[3]{};
     for (int i = 0; i < 64 * 64; ++i)
     {
-      values[i] = float_alpha_to_uint8(new_amaps[alpha_layer + 1][i] * 255.f);
-    }
+        for (int alpha_layer = 0; alpha_layer < nTextures - 1; ++alpha_layer)
+        {
+            values[alpha_layer][i] = float_alpha_to_uint8(new_amaps[alpha_layer + 1][i] * 255.f);
+        }
+        float totVal = (float)(values[0][i]) + values[1][i] + values[2][i];
+        if (totVal > 255)
+        {
+            float mul = (255 / totVal);
+            float vals[3] = { std::floorf(values[0][i] *mul), std::floorf(values[1][i]*mul), std::floorf(values[2][i] *mul)};
```

Comment:
I was slightly confused by this array, so I'd personally just inline it: the values and following lines are independent, so `v = fatu (floor (v  * mul))` seems easier to read.